plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/JdbcPageSink.java

```diff
@@ -161,6 +174,7 @@ else if (javaType == Slice.class) {
                 PreparedStatement statement = this.statement) {
             if (batchSize > 0) {
                 statement.executeBatch();
+                connection.commit();
```

Comment:
@findepi Do we assume that any connection pool or the driver will ensure that the connection's `autoCommit` state it reset before giving the connection to a new client?
Should I explicitly save (in constructor) and restore (here) the autocommit state?

I think this will cause failures with Oracle UCP (it doesn't reset things AFAIK) - let's see.

Reply:
I ask because I remember that for DELETEs Grzegorz is using autocommit (which makes sense because it's always a single statement there) and if the pool gives a stale connection DELETEs will fail.

Reply:
Yep, it failed as expected.

And it absolutely has to do with pool returning non-reset connections because if I remove anything that uses the PageSink by applying the below diff then the test passes.

I'm fixing this by changing the OraclePoolConnectionFactory to always return connections with autoCommit enabled.

<details><summary>Click here for diff</summary>
```diff
diff --git a/plugin/trino-oracle/src/test/java/io/trino/plugin/oracle/TestOraclePoolConnectorSmokeTest.java b/plugin/trino-oracle/src/test/java/io/trino/plugin/oracle/TestOraclePoolConnectorSmokeTest.java
index 30d6108283..e8a38c8c75 100644
--- a/plugin/trino-oracle/src/test/java/io/trino/plugin/oracle/TestOraclePoolConnectorSmokeTest.java
+++ b/plugin/trino-oracle/src/test/java/io/trino/plugin/oracle/TestOraclePoolConnectorSmokeTest.java
@@ -13,15 +13,20 @@
  */
 package io.trino.plugin.oracle;
 
+import com.google.common.collect.ImmutableList;
 import com.google.common.collect.ImmutableMap;
 import io.airlift.testing.Closeables;
 import io.trino.testing.QueryRunner;
+import io.trino.testing.sql.TestTable;
 import org.testng.annotations.AfterClass;
+import org.testng.annotations.Test;
 
 import java.io.IOException;
 
 import static io.trino.plugin.oracle.TestingOracleServer.TEST_PASS;
 import static io.trino.plugin.oracle.TestingOracleServer.TEST_USER;
+import static io.trino.testing.TestingConnectorBehavior.SUPPORTS_CREATE_TABLE;
+import static io.trino.testing.TestingConnectorBehavior.SUPPORTS_DELETE;
 
 public class TestOraclePoolConnectorSmokeTest
         extends BaseOracleConnectorSmokeTest
@@ -44,7 +49,7 @@ public class TestOraclePoolConnectorSmokeTest
                         .put("oracle.connection-pool.enabled", "true")
                         .put("oracle.remarks-reporting.enabled", "false")
                         .build(),
-                REQUIRED_TPCH_TABLES);
+                ImmutableList.of());
     }
 
     @AfterClass(alwaysRun = true)
@@ -54,4 +59,16 @@ public class TestOraclePoolConnectorSmokeTest
         Closeables.closeAll(oracleServer);
         oracleServer = null;
     }
+
+    @Test
+    public void testDeleteAllDataFromTable()
+    {
+        skipTestUnless(hasBehavior(SUPPORTS_CREATE_TABLE) && hasBehavior(SUPPORTS_DELETE));
+        try (TestTable table = new TestTable(oracleServer::execute, "test_delete", "AS SELECT 'good' foo FROM DUAL")) {
+            // not using assertUpdate as some connectors provide update count and some do not
+            assertQuery("SELECT count(*) FROM " + table.getName(), "VALUES 1");
+            getQueryRunner().execute("DELETE FROM " + table.getName());
+            assertQuery("SELECT count(*) FROM " + table.getName(), "VALUES 0");
+        }
+    }
 }
```
</summary>

Reply:
When was playing with Oracle, I noticed that changing a state of the connection survives in a connection in the pool.

Reply:
> Do we assume that any connection pool or the driver will ensure that the connection's `autoCommit` state it reset before giving the connection to a new client?

That should be guaranteed by the pool.

Anything else is very error-prone.

> When was playing with Oracle, I noticed that changing a state of the connection survives in a connection in the pool.

Was this about autocommit state, or something else?

Reply:
I observed it for autocommit state. Hence the 2nd fixup commit.

Hikari does guarantee it - Oracle UCP seems not to.